### DISCUSSION TITLE
Resolve a `TsTypeRef` to the actual type

### DISCUSSION THREAD

**Discussion by invakid404** [swc-project/swc#6242] [Q&A]
Hey there 👋🏻 

I'm in the process of writing my first SWC plugin in Rust, and I'm wondering if there's a way to resolve a `TsTypeRef` (in particular, a `type_param` of a `CallExpr`) to the actual type declaration? I need to extract the fields and methods of the type and generate extra code based on them, but I can't seem to find a way to do it.

Does SWC even keep track of types somewhere? If not, is there a better solution than trying to keep track of type declarations myself?

**Comment by kdy1** [comment]
No. It requires full TypeScript port

**Reply by invakid404** [reply]
What exactly does that mean? Does it mean I have to migrate away from SWC as a whole?
